pkg/bundle/source.go

```diff
@@ -99,27 +99,20 @@ func (b *bundle) buildSourceBundle(ctx context.Context, bundle *trustapi.Bundle)
 			return bundleData{}, fmt.Errorf("failed to retrieve bundle from source: %w", err)
 		}
 
-		opts := util.ValidateAndSanitizeOptions{FilterExpired: b.Options.FilterExpiredCerts}
-		sanitizedBundle, err := util.ValidateAndSanitizePEMBundleWithOptions([]byte(sourceData), opts)
+		err = util.ValidateAndSanitizePEMBundleWithOptions(certPool, []byte(sourceData))
 		if err != nil {
 			return bundleData{}, fmt.Errorf("invalid PEM data in source: %w", err)
 		}
 
-		bundles = append(bundles, string(sanitizedBundle))
 	}
 
 	// NB: empty bundles are not valid so check and return an error if one somehow snuck through.
 
-	if len(bundles) == 0 {
+	if certPool.GetCertsQuantity() == 0 {
```

Comment:
I am not a big fan of this function (name). Would it make sense making the CertPool struct equivalent to a slice (of `certificates` )- so we still can use len?

Reply:
In this case we should export `certificate` field  to be visible in other packages. I guess it is not secure way. 
Am I right?

Reply:
I am not sure if my suggestion even will work. I am withdrawing it. 😉

Reply:
Yes, we can try.
We can do it in separate PR, of course.